src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || array != empty_arraydata) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return array = Arrays.copyOf(array, newCapacity);
+        } else {
+            return array = (T[]) new Object[def_capacity];
+        }
```

Comment:
Few things here:
- `oldCapacity` is redundant. We can just reffer to array.length.
- `multiplicator` is ambigous. What does it multiply? Extract it to class level constant with more descriptive name like `RESIZE_FACTOR`.
- Redundant conditional statement. This method is named `grow`, so it should perform it action always when called. 
![image](https://github.com/kacper-kukula/my-test-task2/assets/105852806/5edd8e5e-af3c-48e4-a588-757d7dac323b)
- No need to add 1 to `newCapacity`, we already multiplited it's size.
- Try to copy your current array to a new, bigger one. Then overwrite the old one with the new one.